runtime/src/main/java/com/flipkart/flux/controller/WorkFlowExecutionController.java

```diff
@@ -493,10 +496,16 @@ private void executeStates(StateMachine stateMachine, Set<State> executableState
                     // register the Task with the redriver
                     // Delay between retires is exponential (2, 4, 8, 16, 32.... seconds) as seen in AkkaTask.
                     // Redriver interval is set as 2 x ( 2^(retryCount+1) x 1s + (retryCount+1) x timeout)
-                    long redriverInterval = 2 * ((int) Math.pow(2, state.getRetryCount() + 1) * 1000 + (state.getRetryCount() + 1) * state.getTimeout());
+                    long redriverInterval;
+                    if (redriverTriggered && state.getStatus() == Status.initialized) {
+                        redriverInterval = 2 * ((int) Math.pow(2, 7) * 1000);
```

Comment:
For 512 seconds, it should be:
redriverInterval = 2 * ((int) Math.pow(2, 8) * 1000);